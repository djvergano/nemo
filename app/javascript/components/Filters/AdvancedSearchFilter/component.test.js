import React from 'react';
import { shallow } from 'enzyme';

import { getFiltersStore } from '../testUtils';

import Component from './component';

const defaultProps = {
  filtersStore: getFiltersStore(),
  onSubmit: jest.fn(),
  onClear: jest.fn(),
};

function withQueryParam() {
  window.location.search = '?search=foo';
}

function withoutQueryParam() {
  window.location.search = '?search=';
}

it('renders as expected (without query param)', () => {
  withoutQueryParam();
  const wrapper = shallow(<Component {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders as expected (with query param)', () => {
  withQueryParam();
  const wrapper = shallow(<Component {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders as expected (with info button)', () => {
  const wrapper = shallow(<Component {...defaultProps} renderInfoButton />);
  expect(wrapper).toMatchSnapshot();
});

describe('callbacks', () => {
  withQueryParam();
  const wrapper = shallow(<Component {...defaultProps} />);

  it('submits', () => {
    wrapper.find('Button.btn-apply').simulate('click');
    expect(defaultProps.onSubmit).toMatchSnapshot();
  });
});
