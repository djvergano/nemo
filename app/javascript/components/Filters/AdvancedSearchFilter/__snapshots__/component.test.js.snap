// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`callbacks submits 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`renders as expected (with info button) 1`] = `
<div
  className="d-flex"
>
  <i
    className="fa fa-search"
  />
  <input
    autoComplete="off"
    className="form-control search-str"
    name="search"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search"
    type="text"
    value="query"
  />
  <Button
    className="btn-apply btn-advanced-search"
    onClick={[MockFunction]}
    variant="secondary"
  >
    Search
  </Button>
  <i
    className="fa fa-info-circle hint"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  />
</div>
`;

exports[`renders as expected (with query param) 1`] = `
<div
  className="d-flex"
>
  <i
    className="fa fa-search"
  />
  <input
    autoComplete="off"
    className="form-control search-str"
    name="search"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search"
    type="text"
    value="query"
  />
  <Button
    className="btn-apply btn-advanced-search"
    onClick={[MockFunction]}
    variant="secondary"
  >
    Search
  </Button>
</div>
`;

exports[`renders as expected (without query param) 1`] = `
<div
  className="d-flex"
>
  <i
    className="fa fa-search"
  />
  <input
    autoComplete="off"
    className="form-control search-str"
    name="search"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="Search"
    type="text"
    value="query"
  />
  <Button
    className="btn-apply btn-advanced-search"
    onClick={[MockFunction]}
    variant="secondary"
  >
    Search
  </Button>
</div>
`;
